opm/simulators/linalg/ExtraSmoothers.hpp

```diff
@@ -0,0 +1,42 @@
+#ifndef OPM_EXTRASMOOTHERS_HPP
+#define OPM_EXTRASMOOTHERS_HPP
+
+#include "DILU.hpp"
+
+namespace Dune
+{
+  template <class M, class X, class Y>
+    class SeqDilu;
+
+namespace Amg
+{
+    /**
+     * @brief Policy for the construction of the SeqDilu smoother
+     */
+    template <class M, class X, class Y>
+    struct ConstructionTraits<SeqDilu<M, X, Y>> {
+        typedef DefaultConstructionArgs<SeqDilu<M, X, Y>> Arguments;
+#if DUNE_VERSION_NEWER(DUNE_ISTL, 2, 7)
+        static inline std::shared_ptr<SeqDilu<M, X, Y>> construct(Arguments& args)
+        {
+            return std::make_shared<SeqDilu<M, X, Y>>(
+							  args.getMatrix());
```

Comment:
Weird formatting, also on line 30. This is due to the presence of tabs in the file. Avoid tabs in files, make your editor always output as spaces only.

Reply:
My editor uses spaces, but it seems like something weird happened here. Fixed it now.